_posts/2019-05-02-zarr-2.3-release.md

```diff
@@ -0,0 +1,134 @@
+interface means it is relatively straightforward to add support for a
+range of different storage systems. The 2.3 release adds support for
+storage using [SQLite](@@TODO), [Redis](@@TODO), [MongoDB](@@TODO) and
+[Azure Blob Storage](@@TODO).
+
+For example, here's code that stores an array using Redis:
+
+{% highlight python %}
+TODO
+{% endhighlight %}
+
+Here's the same example but storing the data in the cloud via Azure
+Blob Storage:
+
+{% highlight python %}
+TODO
+{% endhighlight %}
+
+Support for other cloud object storage storage services was already
+available via other packages, with Amazon S3 supported via the
+[s3fs](@@TODO) package, and Google Cloud Storage supported via the
+[gcsfs](@@TODO) package.
+
+The attraction of cloud storage is that total I/O bandwidth scales
+linearly with the size of a computing cluster, so there are no
+technical limits to the size of the data or computation you can scale
+up to. Here's a slide from a recent presentation from Ryan Abernathy
+showing how I/O scales when using Zarr, and comparing that to reading
+data from a remote data service which does not scale in the same way:
+
+@@TODO plot
+
+## Optimisations for cloud storage: consolidated metadata
+
+One issue with using cloud object storage is that, although total I/O
+throughput can be high, the latency involved in each request to read
+the contents of an object can be around 100 ms, even when reading from
+compute nodes within the same data centre. This latency can add up
+when reading many arrays, because in Zarr each array has its own
+metadata stored in a separate object.
+
+To work around this, the 2.3 release adds an experimental feature to
+consolidate metadata for all arrays and groups within a hierarchy into
+a single object, which can be read once via a single request. Although
+this is not suitable for rapidly changing datasets, it can be good for
+large datasets which are relatively static.
+
+To use this feature, two new convenience functions have been
+added. The [`consolidate_metadata()`](@@TODO) function performs the
+initial consolidation, reading all metadata and combining them into a
+single object. Once you have done that and deployed the data to a
+cloud object store, the [`open_consolidated()`] function can be used
+to read data, making use of the consolidated metadata.
+
+Support for this new feature is also now available via the
+[xarray](@@TODO) and [intake](@@TODO), and the Pangeo project is using
+consolidated metadata for all new Zarr datasets. E.g., here's an
+example of how to open a Zarr dataset from Pangeo's data catalog via
+intake:
+
+{% highlight python %}
+TODO
+{% endhighlight %}
+
+## Compatibility with N5
+
+Around the same time that development on Zarr was getting started, a
+separate team led by Stephan Saafeld at the Janelia research campus
+was experiencing similar frustrations trying to use HDF5 for storing
+brain imaging data, and developed a software library called
```

Comment:
Electron microscopy data? Also IDK how I feel about "brain" as it feels non-technical, but I don't know how much technical verbage we want to include here. @axtimwalde, thoughts?

Reply:
I was hoping you and/or @axtimwalde could improve this :-)